runtime/vm/swalk.c

```diff
@@ -178,393 +153,155 @@ UDATA  walkStackFrames(J9VMThread *currentThread, J9StackWalkState *walkState)
 	walkState->stackMap = NULL;
 	walkState->inlineMap = NULL;
 	walkState->inlinedCallSite = NULL;
+	walkState->privateFlags = walkState->walkThread->privateFlags;
+	walkState->stackObject = walkState->walkThread->stackObject;
+	walkState->stackWalkState = walkState->walkThread->stackWalkState;
 #ifdef J9VM_INTERP_NATIVE_SUPPORT
 	walkState->jitInfo = NULL;
 	walkState->inlineDepth = 0;
 	walkState->inlinerMap = NULL;
-	walkState->walkedEntryLocalStorage = walkState->walkThread->entryLocalStorage;
-	walkState->i2jState = walkState->walkedEntryLocalStorage ? &(walkState->walkedEntryLocalStorage->i2jState) : NULL;
 	walkState->j2iFrame = walkState->walkThread->j2iFrame;
+	J9VMEntryLocalStorage* els = walkState->walkThread->entryLocalStorage;
+	walkState->walkedEntryLocalStorage = els;
+	if (NULL != els) {
+		walkState->i2jState = &(els->i2jState);
+		walkState->jitGlobalStorageBase = els->jitGlobalStorageBase;
+		walkState->jitFPRegisterStorageBase = els->jitFPRegisterStorageBase;
+		walkState->oldEntryLocalStorage = els->oldEntryLocalStorage;
+	} else {
+		walkState->i2jState = NULL;
+		walkState->jitGlobalStorageBase = NULL;
+		walkState->jitFPRegisterStorageBase = NULL;
+		walkState->oldEntryLocalStorage = NULL;
+	}
 #ifdef J9VM_JIT_FULL_SPEED_DEBUG
 	walkState->decompilationStack = walkState->walkThread->decompilationStack;
 	walkState->decompilationRecord = NULL;
 	walkState->resolveFrameFlags = 0;
-#endif
-#endif
+#endif /* J9VM_JIT_FULL_SPEED_DEBUG */
+#endif /* J9VM_INTERP_NATIVE_SUPPORT */
 
+resumeInterpreterWalk:
```

Comment:
Maybe ` resumeWalk` here instead of `resumeInterpreterWalk` as both interpreter / JIT walk are possible.